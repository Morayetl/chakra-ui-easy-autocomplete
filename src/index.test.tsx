import React from 'react';
import renderer from 'react-test-renderer';
import { ChakraUIEasyAutoComplete, ChakraUIEasyAutoCompleteSuggestionItemProps } from '.'
import { Text } from '@chakra-ui/react';

const suggestions: Array<ChakraUIEasyAutoCompleteSuggestionItemProps> = [
  {
    label: 'cow',
    value: 'moow'
  },
  {
    label: 'chicken',
    value: 'kotkot'
  },
  {
    label: 'pig',
    value: 'röff röff'
  }
]

describe('Renders', () => {
  it('correctly', () => {
    const component = renderer.create(
      <ChakraUIEasyAutoComplete
        size="sm"
        w="200px"
        suggestions={suggestions}
        onChange={(e: any) => console.log(e.target.value)}
        resultNotFoundElement={<Text p={3}>Result not found</Text>}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  })
})
