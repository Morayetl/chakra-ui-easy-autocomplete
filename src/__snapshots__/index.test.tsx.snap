// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Array [
  <div
    className="css-0"
  >
    <div
      className="chakra-input__group css-1t5oxgh"
      onChange={[Function]}
      suggestions={
        Array [
          Object {
            "label": "cow",
            "value": "moow",
          },
          Object {
            "label": "chicken",
            "value": "kotkot",
          },
          Object {
            "label": "pig",
            "value": "röff röff",
          },
        ]
      }
    >
      <input
        className="chakra-input css-1bzjvz1"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        suggestions={
          Array [
            Object {
              "label": "cow",
              "value": "moow",
            },
            Object {
              "label": "chicken",
              "value": "kotkot",
            },
            Object {
              "label": "pig",
              "value": "röff röff",
            },
          ]
        }
        type="text"
        value=""
      />
    </div>
  </div>,
  <div
    className="css-cjymd2"
    onChange={[Function]}
    size="sm"
    suggestions={
      Array [
        Object {
          "label": "cow",
          "value": "moow",
        },
        Object {
          "label": "chicken",
          "value": "kotkot",
        },
        Object {
          "label": "pig",
          "value": "röff röff",
        },
      ]
    }
  />,
  <span />,
]
`;
